main.py

```diff
@@ -72,17 +73,20 @@
 
     bot_view = BotViewV1(bot_controller, bot_service, auth_controller)
 
+
     data_source_view = DataSourceViewV1(auth_controller)
     engine_selector = ChatEngineSelector(
         openai_api_key=config.openai_api_key, anthropic_api_key=config.anthropic_api_key
     )
     
     document_repository = PostgresDocumentRepository(sessionmaker)
 
-    document_service = DocumentServiceV1(document_repository)
+    document_service = DocumentServiceV1(config.aws_access_key_id, config.aws_secret_access_key, config.aws_public_bucket_name, config.aws_region, document_repository)

```

Comment:
Minor nitpick: since there's 4 values for AWS-related configs, have you considered turning this into a TypedDict or class (like with GoogleCredentials)?

Reply:
Done, thanks for the suggestion